src/components/SearchBar.vue

```diff
@@ -0,0 +1,123 @@
+<template>
+	<section>
+		<label for="userbox">
+			Username(s):
+		</label>
+		<input
+			id="userbox"
+			class="userfield"
+			type="text"
+			placeholder="ex: holopollock, qjack"
+			:value="value"
+			v-on:keyup.enter=action()
+			v-on:input="updateValue($event.target.value)"
+		/>
+		<button 
+			v-on:click=action()
+		>
+			Submit
+		</button>
+		<slot/>
+	</section>
+</template>
+
+<script>
+	export default
+	{
+		name: 'SearchBar',
+		props:
+		[
+			'value',
+			'action'
+		],
+		methods:
+		{
+			updateValue: function (value)
```

Comment:
this looks weird?

Reply:
like the `name: function (value)`-formatting? It was just the way they did it in the Vue docs, I can change it if you'd prefer.